src/js/control-bar/progress-control/seek-bar.js

```diff
@@ -100,7 +105,18 @@ class SeekBar extends Slider {
    * @private
    */
   update_(currentTime, percent) {
-    const duration = this.player_.duration();
+    const liveTracker = this.player_.liveTracker;
+    let duration = this.player_.duration();
+
+    if (liveTracker.isLive()) {
+      duration = this.player_.liveTracker.seekableEnd();
+    }
+
+    if (liveTracker.seekableEnd() === Infinity) {
```

Comment:
I'm not sure if seekable.end is ever `Infinity`

Reply:
actually this is the android thing again, right? If so, a comment will help a lot!

Reply:
it is on Android native hls